docs: refactor docs to enable the local development (#2454)
elastic/e2e-testing: .ci/README.md

The following variables need to be exported:

- *AWS_SECRET_ACCESS_KEY*: AWS secret access key
- *AWS_ACCESS_KEY_ID*: AWS access key id

Install python deps:

```shell
make -C .ci setup-env
```

It will create a `.runID` under the `.ci` directory. It will contain an unique identifier for your machines, which will be added as a VM tag.

We are able to run the Elastic Stack and all the test suites in AWS instances, so whenever a build failed we will be able to recreate the same machine and access it to inspect its state: logs, files, containers... For that, we are enabling SSH access to those ephemeral machines, which can be created following this guide.

But you must first understand that there are two types of Cloud machines: 
1) the VM running the stack, and 

> 2) the VMs where the Elastic Agent will be installed and enrolled into the stack.

Reviewer: [Suggestion]
```suggestion
2) the VMs running the actual tests, where the Elastic Agent will be installed and enrolled into the stack.
```

### Create and configure the stack VM

This specialised VM starts Elasticsearch, Kibana and Fleet Server using Docker Compose, but instead of invoking the compose file directly, it uses the test framework to do it. Why? Because we need to wait for Elasticsearch to be ready and request an API Token to be passed to the Fleet Server container. And [we do this with code](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/e2e/_suites/fleet/fleet.go#L631-L748).

The VM is a Debian AMD64 machine, as described [here](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/.ci/.e2e-platforms.yaml#L3-L7).

```shell

> export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS"

Reviewer: [Suggestion]
If I understood correctly the Makefile, this is optional on unix systems, right?
```suggestion
export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS" # optional, defaults to $(HOME)/.ssh/id_rsa
```

make -C .ci create-stack
```

> A `.stack-host-ip` file will be created in the `.ci` directory of the project including the IP address of the stack instance. Check it out from that file, or make a note of the IP address displayed in the ansible summary, as you'll probably need it to connect your browser to open Kibana, or to SSH into it for troubleshooting.

Reviewer: I know it's a bit lazy, but we could have a `make show-stack-ip` that `cat` this file. What do you think?

Author: Done

Author: ```shell
$ make -C .ci show-stack
Stack IP address    : 18.220.137.250
Stack type          : debian_amd64
Stack instance type : c5.4xlarge
Stack SSH user      : admin
```

> The IP address of the stack in the `.stack-host-ip` file will be used by the automation.

Please remember to [destroy the stack](#destroying-the-stack-and-the-test-node) once you finished your testing.

> ### Create and configure test node

Reviewer: ```suggestion
### Create and configure the test node
```

> There are different VM flavours that you can use to run the Elastic Agent and enrol it into the Stack: Debian, CentOS, SLES15, Oracle Linux... using AMD and ARM as architecture. You can find the full reference of the platform support [here](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/.ci/.e2e-platforms.yaml#L2-L42).

Reviewer 2: ```suggestion
There are different VM flavours that you can use to run the Elastic Agent and enroll it into the Stack: Debian, CentOS, SLES15, Oracle Linux... using AMD and ARM as architecture. You can find the full reference of the platform support [here](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/.ci/.e2e-platforms.yaml#L2-L42).
```

In these VMs, the test framework will download a binary to install the Elastic Agent (TAR files, DEB/RPM packages...), and will execute the different agent commands to install, enroll, uninstall, etc.

It's possible to configure the test node (OS, architecture), using the values that are already present in [the platforms descriptor](.e2e-platforms.yaml):

```shell
# example for Centos 8 ARM 64
export NODE_IMAGE="ami-01cdc9e8306344fe0"

> export NODE_INSTANCE_TYPE="a1.larg"

Reviewer 2: Should this be `a1.large` instead of `a1.larg`?

> export NODE_INSTANCE_TYPE="a1.larg"

Author: ```suggestion
export NODE_INSTANCE_TYPE="a1.large"
```

export NODE_LABEL="centos8_arm64"
export NODE_USER="centos"
```

Besides that, it's possible to configure the test node for the different test suites that are present in the test framework: `fleet`, `helm` and `kubernetes-autodiscover`. Please configure the test node setting the suite, being `fleet` the default:

```shell
# all possible suites
export SUITE="fleet"
export SUITE="helm"
export SUITE="kubernetes-autodiscover"
```

Finally, please create the test node:

> ```shell
> export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS"
> export SUITE="fleet"
> make -C .ci create-node

Reviewer: [Suggestion]
I personally find better to set them inline
```suggestion
```shell
SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS" SUITE="fleet" make -C .ci create-node
```

Author: Me too. I added it separately for clarity, although having them inline would have the user always setting it, which could be positive to avoid wrong values for it.

```

A `.node-host-ip` file will be created in the `.ci` directory of the project including the IP address of the node instance. Check it out from that file, or make a note of the IP address displayed in the ansible summary, as you'll probably need it to SSH into it for troubleshooting.

> The IP address of the node in that file will be used by the automation.

> Please remember to [destroy the node](#destroying-the-stack-and-the-test-node) once you finished your testing.

Reviewer 2: ```suggestion
Please remember to [destroy the node](#destroying-the-stack-and-the-test-node) once you have finished your testing.
```

### Run a test suite

> You can select the specific tags that you want to include in the test execution. Please look for the different tags in the existing feature files for the suite you are interested in running:

Reviewer: [Suggestion]
Show or state where the tags are defined. From my experience, the first time I saw this tags being referenced, I had no idea where they were defined, therefore I did not know what to set them to

Reviewer 2: +1 to this suggestion. I have had this issue in the past.

Author: Indeed. Adding it to the docs:

> You can select the specific tags that you want to include in the test execution. Please look for the different tags in the existing feature files for the suite you are interested in running. For that, please check out the tags/annotations that are present in those feature files (`*.feature`), which live in the `features` directory under your test suite. In example, for `fleet` test suite, you can find them [here](../e2e/_suites/fleet/features/).

> When submitting the pull request with your user to enable the SSH access, please remember to add the right backport labels (ex. `backport-v8.2.0`) so that you will be able to SSH into the CI machines for all supported maintenance branches.

To SSH into the machines, please use the following commads:

```shell
export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS"
make -C .ci ssh-stack
make -C .ci ssh-node
```

### Destroying the Elastic Stack and recreating fleet-server

Sometimes you need to tear down the Elastic Stack, or recreate the fleet-server, mostly in the case the API Token used for Fleet Server [expired after 1 hour](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#token-service-settings).

```shell
export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS"
make -C .ci destroy-elastic-stack
make -C .ci recreate-fleet-server
```

### Destroying the stack and the test nodes

> Do not forget to destroy the stack and nodes you use!

Reviewer: ```suggestion
Do not forget to destroy the stack and nodes once you're done with your tests!
```
